File: playground/hoa0217/java-springboot-order-perf-optimizer/src/main/java/com/order/perf/dto/OrderItemDto.java
package com.order.perf.dto;

import com.order.perf.domain.OrderStatus;
import lombok.Getter;

import java.time.LocalDateTime;

// 주문 항목 DTO
@Getter
public class OrderItemDto {

    private final Long orderId;
    private int orderNumber;
    private final OrderStatus orderStatus;
    private final LocalDateTime createdAt;
    private final LocalDateTime updatedAt;

    public OrderItemDto(final Long orderId, final int orderNumber,
                        final OrderStatus orderStatus,
                        final LocalDateTime createdAt, final LocalDateTime updatedAt) {
        this.orderId = orderId;
        this.orderNumber = orderNumber;
        this.orderStatus = orderStatus;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }
}

File: playground/ruthetum/app/e-commerce/order/src/main/kotlin/team/sipe/spurt/ecommerce/order/dto/OrderItemDto.kt
package team.sipe.spurt.ecommerce.order.dto

import java.time.LocalDateTime

data class OrderItemDto(
    val orderId: Long,
    val orderNumber: Long,
    val orderStatus: team.sipe.spurt.ecommerce.order.domain.OrderStatus,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
) {
    companion object {
        fun of(order: team.sipe.spurt.ecommerce.order.domain.Order): team.sipe.spurt.ecommerce.order.dto.OrderItemDto {
            return team.sipe.spurt.ecommerce.order.dto.OrderItemDto(
                orderId = order.id!!,
                orderNumber = order.orderNumber,
                orderStatus = order.orderStatus,
                createdAt = order.createdAt,
                updatedAt = order.updatedAt,
            )
        }
    }
}